shared/middleware/files_api.rb

```diff
@@ -206,12 +250,44 @@ def put_file(endpoint, encrypted_channel_id, filename, body)
     put_file('assets', encrypted_channel_id, file[:filename], file[:tempfile].read)
   end
 
+  # POST /v3/animations/<channel-id>/<filename>?version=<version-id>
+  #
+  # Create or replace an animation. We use this method so that IE9 can still
+  # upload by posting to an iframe.
+  #
+  post %r{/v3/(animations)/([^/]+)/([^/]+)$} do |endpoint, encrypted_channel_id, filename|
```

Comment:
It's not clear that POSTing rather than PUTting (which would be more RESTful) is truly necessary here, since the uploads should be same-origin. Probably a good time to address this is after we stop supporting IE9. No action required now as grepping for IE9 will find this comment.